src/main/java/entities/Tag.java

```diff
@@ -0,0 +1,42 @@
+package entities;
+
+import java.util.ArrayList;
+
+public class Tag {
+    // The name of the tag
+    final String name;
+
+    // Array of posts that are under this tag
+    final ArrayList<Post> taggedPosts;
+
+    /**
+     * Initializes an instance of Tag.
+     *
+     * @param name The name of this tag.
+     */
+    public Tag(String name) {
+        this.name = name;
+        this.taggedPosts = new ArrayList<>();
+    }
+
+    /**
+     * Add a post that has this tag.
+     *
+     * @param post The post that has this name as a tag.
+     */
+    public void addTag(Post post) {
+        taggedPosts.add(post);
+    }
```

Comment:
should we check if the post is already in the tags list (is it possible we could fuck that up somewhere in the code?) and throw an exception?